packages/filesystem/src/browser/file-resource.ts

```diff
@@ -242,4 +272,14 @@ export class FileResourceResolver implements ResourceResolver {
         return !!await dialog.open();
     }
 
+    protected async shouldOpenAsText(uri: URI, error: string): Promise<boolean> {
+        const dialog = new ConfirmDialog({
```

Comment:
Why do not we do what VS Code does and open an editor with the message instead of modal dialogs?

https://github.com/microsoft/vscode/blob/a6eccc39f84a274ec577fb265195d290e16799d4/src/vs/workbench/browser/parts/editor/binaryEditor.ts#L207-L227

Reply:
It is not possible for us to return another widget for `EditorManager.open` method.

Reply:
OK.